com.microsoft.mrtk.uxcore/Dialog/DialogSpawner.cs

```diff
@@ -0,0 +1,143 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using TMPro;
+using UnityEngine;
+using UnityEngine.Serialization;
+using UnityEngine.Events;
+
+namespace Microsoft.MixedReality.Toolkit.UX
+{
+    /// <summary>
+    /// Spawns dialogs with the requested parameters and manages
+    /// the lifecycle of the resulting dialog component.
+    /// </summary>
+    [ExecuteAlways]
+    [AddComponentMenu("MRTK/UX/Dialog Spawner")]
+    public class DialogSpawner : MonoBehaviour
+    {
+        /// <summary>
+        /// Specifies the <see cref="DialogSpawner"/>'s behavior
+        /// when opening a dialog while one is already active.
+        /// </summary>
+        public enum Policy
+        {
+            /// <summary> Dismisses any existing dialog, and then spawns its own. </summary>
+            DismissExisting,
+
+            /// <summary> Aborts spawning if a dialog is already active. </summary>
+            AbortIfExisting
+        }
+
+        /// <summary>
+        /// The default prefab to instantiate when spawning a dialog.
+        /// </summary>
+        [field: SerializeField, Tooltip("The default prefab to instantiate when spawning a dialog.")]
+        public GameObject DialogPrefab { get; set; }
+
+        // Static reference to the dialog instance, if active and spawned.
+        // Will be null when no dialog is open.
+        private static IDialog dialogInstance = null;
+
+        // Previously-dismissed dialogs pooled by type.
+        private static Dictionary<Type, Queue<IDialog>> dialogPool = new Dictionary<Type, Queue<IDialog>>();
+
+        /// <summary>
+        /// Retrieves or creates a new dialog instance. Specify a <paramref name="spawnPolicy"/> to
+        /// determine how existing dialogs are handled. By default, the spawner will use its <see cref="DialogPrefab"/>,
+        /// but a custom <paramref name="prefab"/> can also be specified. Dialogs of different types
+        /// will be pooled independently. Only one dialog can be visible at a time, globally.
+        /// </summary>
+        /// <remarks>
+        /// To build your dialog, call the fluent builder methods on the <see cref="IDialog"/> instance retuirned
+        /// by this method.
+        /// </remarks>
+        /// <param name="spawnPolicy">How the spawner should deal with existing dialogs (dismiss or abort)</param>
+        /// <param name="prefab">The prefab to use for the dialog. If null, the spawner's <see cref="DialogPrefab"/> will be used.</param>
+        /// <returns>A dialog instance, or null if the spawner was unable to spawn a dialog.</returns>
+        public IDialog Get(Policy spawnPolicy = Policy.DismissExisting, GameObject prefab = null)
```

Comment:
nit. calling this `Spawn` or `SpawnDialog` is a bit more intuitive compared to `Get`

Reply:
I went back and forth on this a bit (see the first comment from Nick)

Ideally, the prefab isn't spawned at all until a `.Build()` or `.Show()` method at the end of the fluent method chain. But there needs to be an initial call that actually grabs an instance from the pool. I wanted to avoid calling it "Spawn" because that implies that it actually shows a dialog; it doesn't, nothing is visible/hydrated/bound until the rest of the methods are chained and `.Show()` is called to finalize.

Reply:
Calling `Get()` something like `Build()` instead could be interesting, but that's confusing because typical fluent style APIs would put Build at the end.

Reply:
Switched naming to `DialogPool.Get()`.